### PR TITLE
Include `participant_sid` in `UpdateSubscription`.

### PR DESCRIPTION
**Pull request by boks1971**
Prevents all publisher tracks to find a match.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -82,3 +82,5 @@ require (
 	google.golang.org/grpc v1.42.0 // indirect
 	gopkg.in/square/go-jose.v2 v2.5.1 // indirect
 )
+
+replace github.com/livekit/protocol => ../protocol
```
Comment (boks1971) [inline go.mod:86]: Will update once protocol is merged and versioned.

File: pkg/rtc/room.go [hunk 2]
```diff
@@ -92,6 +92,19 @@ func (r *Room) GetParticipant(identity string) types.Participant {
 	return r.participants[identity]
 }
 
+func (r *Room) GetParticipantBySid(participantID string) types.Participant {
```
Comment (boks1971) [inline pkg/rtc/room.go:95]: Sooner or later, looks like we will need a fast look up of participant in room by either identity or sid. So, maybe we need to maintain two maps.

File: pkg/rtc/room.go [hunk 3]
```diff
@@ -322,16 +335,34 @@ func (r *Room) RemoveParticipant(identity string) {
...
-				}
+	for _, trackSid := range trackIds {
+		for _, p := range participants {
+			track := p.GetPublishedTrack(trackSid)
+			if track != nil {
+				tracks[trackSid] = track
+				break
+			}
+		}
+	}
```
Comment (boks1971) [inline pkg/rtc/room.go:355]: optimising this a bit by asking participant for specific track sid and breaking as soon as one is found.

File: pkg/rtc/room.go [hunk 4]
```diff
@@ -322,16 +335,34 @@ func (r *Room) RemoveParticipant(identity string) {
...
+
+	for _, pt := range participantTracks {
+		p := r.GetParticipantBySid(pt.ParticipantSid)
```
Comment (boks1971) [inline pkg/rtc/room.go:358]: This is where we need look up room by participant sid.

File: pkg/rtc/room.go [hunk 5]
```diff
@@ -322,16 +335,34 @@ func (r *Room) RemoveParticipant(identity string) {
...
+		if p == nil {
+			continue
+		}
+		for _, trackSid := range pt.TrackSids {
+			track := p.GetPublishedTrack(trackSid)
+			if track != nil {
+				tracks[trackSid] = track
```
Comment (boks1971) [inline pkg/rtc/room.go:365]: Layering the new field on top and keep building the tracks list.

File: pkg/rtc/room.go (hunk 2 above)
Comment (davidzhao) [reply pkg/rtc/room.go:95]: I think a identity to sid map makes a ton of sense. and keying off of sid internally.

**Review by davidzhao — APPROVED** [review]
looks good!

**Pull request merged**